Mathlib/Algebra/Category/GroupCat/Preadditive.lean

```diff
@@ -0,0 +1,45 @@
+/-
+Copyright (c) 2020 Markus Himmel. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Markus Himmel
+
+! This file was ported from Lean 3 source module algebra.category.Group.preadditive
+! leanprover-community/mathlib commit 829895f162a1f29d0133f4b3538f4cd1fb5bffd3
+! Please do not edit these lines, except to modify the commit id
+! if you have ported upstream changes.
+-/
+import Mathlib.Algebra.Category.GroupCat.Basic
+import Mathlib.CategoryTheory.Preadditive.Basic
+
+/-!
+# The category of additive commutative groups is preadditive.
+-/
+
+open CategoryTheory
+
+universe u
+
+namespace AddCommGroupCat
+
+-- porting note: this instance was not necessary in mathlib
+instance (P Q : AddCommGroupCat) : AddCommGroup (P ⟶ Q) :=
+  (inferInstance : AddCommGroup (AddMonoidHom P Q))
+
+-- porting note: this lemma was not necessary in mathlib
+@[simp]
+lemma hom_add_apply {P Q : AddCommGroupCat} (f g : P ⟶ Q) (x : P) :
+  (f + g) x = f x + g x := rfl
+
+section
+
+-- porting note: the simp attribute was locally deactivated here,
+-- otherwise Lean would try to infer `Preadditive AddCommGroupCat`
+-- in order to prove the axioms `add_comp` and `comp_add` in the
+-- next instance declaration
+attribute [-simp] Preadditive.add_comp Preadditive.comp_add
```

Comment:
```suggestion
attribute [-simp] Preadditive.add_comp Preadditive.comp_add
```
Is this locally or globally deactivating `simp`?

Reply:
I have struggled with the syntax (trying `local attribute...`, `attribute [local -simp]`, or `attribute [-local simp]`, but it seems that it is deactivating `simp` only in this `section...end` block. Just in case, you may try to insert these lines before or after the `end`:
```lean
example {C : Type _} [Category C] [Preadditive C] {X Y Z : C}
    (f f' : X ⟶ Y) (g : Y ⟶ Z) : (f + f') ≫ g = f ≫ g + f' ≫ g :=
  by simp?
```